src/plugins/notebook/components/NotebookEntry.vue

```diff
@@ -205,13 +216,45 @@ export default {
             }
         }
     },
+    data() {
+        return {
+            editMode: false,
+            canEdit: true
+        };
+    },
     computed: {
         createdOnDate() {
             return this.formatTime(this.entry.createdOn, 'YYYY-MM-DD');
         },
         createdOnTime() {
             return this.formatTime(this.entry.createdOn, 'HH:mm:ss');
         },
+        formattedText() {
+            let text = this.entry.text;
+
+            if (this.editMode) {
+                return sanitizeHtml(text);
+            }
+
+            let urlsExist = false;
+            let urlMap = {};
+
+            while (text.match(URL_REGEX)) {
+                let id = uuid();
+
+                urlsExist = true;
+                urlMap[id] = text.match(URL_REGEX)[0];
+                text = text.replace(urlMap[id], id);
+            }
+
+            if (urlsExist) {
+                for (const [id, url] of Object.entries(urlMap)) {
+                    text = text.replace(id, `<a class="c-hyperlink" target="_blank" href="${url}">${url}</a>`);
+                }
+            }
+
+            return sanitizeHtml(text, SANITIZATION_SCHEMA);
```

Comment:
We might be able to make this even more secure, and save some configuration, if we just sanitize the whole thing first, before we do anything, with no exclusions.

Then, in step 2, we replace all the URLs with an anchor tag that we define ourselves. This way there are no custom anchor tags getting through at all.

Reply:
Yeah, I like that. I'll set that up.